src/resonanceReconstruction/rmatrix/ChannelQuantumNumbers/src/toString.hpp

```diff
@@ -0,0 +1,23 @@
+/**
+ *  @brief Produce a string identifier for the set of quantum numbers 
+ *
+ *  This function produces a string identifier for the set of quantum numbers
+ *  formatted like this: {l,s,Jpi} in which l values are integers, s and J
+ *  half integers and pi either + or -.
+ */
+std::string toString() const {
+
+  auto toHalfIntegerString = 
+      [] ( const double a )
+         { double half;
+           return std::modf( a, &half ) == 0. ?
+                      std::to_string( static_cast<int>( half ) ) :
+                      std::to_string( 2 * static_cast<int>( half ) + 1 ) 
+                        + "/2"; };
+
+  return "{" + std::to_string( this->l_ ) + ","
+             + toHalfIntegerString( this->s_ ) + ","
+             + toHalfIntegerString( this->J_ )
+             + ( this->parity_ > 0 ? "+" : "-" ) + "}";
+}
```

Comment:
Including a complete dependency just for a single format statement isn't worth it to me. I'll change it when we move to C++20 and it's part of the standard :-)

And yes, I know it is an implicit dependency because of ENDFtk and Log.